improver/psychrometric_calculations/hail_size.py

```diff
@@ -419,27 +479,62 @@ def get_hail_size(self, vertical: np.ndarray, horizontal: np.ndarray) -> np.ndar
         # flattens array's so they can later be accessed
         horizontal_flat = horizontal_rounded.flatten(order="C")
         vertical_flat = vertical_rounded.flatten(order="C")
+        wet_bulb_zero_flat = wet_bulb_zero.flatten(order="C")
+
         hail_size_list = []
         # clips index values to not be longer than the table
         vertical_clipped = np.clip(vertical_flat, None, len(lookup_table) - 1)
         horizontal_clipped = np.clip(horizontal_flat, None, len(lookup_table[0]) - 1)
 
-        for vert, hor in zip(vertical_clipped, horizontal_clipped):
-            if min(hor, vert) < 0 or not (vert and hor):
-                hail_size_list.append(0)
+        for vert, hor, wbz in zip(
+            vertical_clipped, horizontal_clipped, wet_bulb_zero_flat
+        ):
+            if min(hor, vert) < 0 or not (vert and hor) or wbz > 4400:
+                hail_size = 0
+            elif wbz < 3300:
+                hail_size = lookup_table[int(vert)][int(hor)]
```

Comment:
I would rearrange this to separate the two concerns. First, calculate the hail size from the first nomogram. Then, `if wbz >= 3300` - reduce it.
```suggestion
            if min(hor, vert) < 0 or not (vert and hor) or wbz > 4400:
                hail_size = 0
            else:
                hail_size = lookup_table[int(vert)][int(hor)]
```

Reply:
updated